soccer/planning/planner/CollectPlanner.cpp

```diff
@@ -0,0 +1,139 @@
+#include "CollectPlanner.hpp"
+#include "planning/planner/PlanRequest.hpp"
+#include "planning/trajectory/RRTUtil.hpp"
+namespace Planning {
+ConfigDouble* CollectPlanner::_bufferDistBeforeContact;
+ConfigDouble* CollectPlanner::_bufferDistAfterContact;
+ConfigDouble* CollectPlanner::_ballSpeedApproachDirectionCutoff;
+ConfigDouble* CollectPlanner::_touchDeltaSpeed;
+ConfigDouble* CollectPlanner::_targetChangeThreshold;
+ConfigDouble* CollectPlanner::_controlAccelScalePercent;
+ConfigDouble* CollectPlanner::_distCutoffToControl;
+
+REGISTER_CONFIGURABLE(CollectPlanner);
+
+void CollectPlanner::createConfiguration(Configuration* cfg) {
+    _bufferDistBeforeContact =
+        new ConfigDouble(cfg, "Capture/Collect/bufferDistBeforeContact");
+    _bufferDistAfterContact =
+        new ConfigDouble(cfg, "Capture/Collect/bufferDistAfterContact");
+    _ballSpeedApproachDirectionCutoff = new ConfigDouble(
+        cfg, "Capture/Collect/ballSpeedApproachDirectionCutoff");
+    _touchDeltaSpeed = new ConfigDouble(cfg, "Capture/Collect/touchDeltaSpeed");
+    _targetChangeThreshold =
+        new ConfigDouble(cfg, "Capture/Collect/targetChangeThreshold");
+    _controlAccelScalePercent =
+        new ConfigDouble(cfg, "Capture/Collect/controlAccelScalePercent");
+    _distCutoffToControl =
+        new ConfigDouble(cfg, "Capture/Collect/distCutoffToControl");
+}
+
+using namespace Geometry2d;
+Trajectory CollectPlanner::plan(PlanRequest&& request) {
+    processStateTransitions(request);
+    switch (_collectStates[request.shellID]) {
+        case CollectState::Course:
+            return course(std::move(request));
+        case CollectState::Fine:
+            return fine(std::move(request));
+        case CollectState::Control:
+            return control(std::move(request));
+    }
+    debugThrow("Error: CollectPlanner state machine failed.");
+    return Trajectory{{}};
+}
+Trajectory CollectPlanner::course(PlanRequest&& request) {
+    RobotInstant startInstant = request.start;
+    RotationConstraints rotationConstraints = request.constraints.rot;
+    const Ball& ball = request.context->state.ball;
+    Point approachDir = findApproachDirection(request);
+    Point noisyTargetPoint =
+        ball.pos - approachDir * (Ball_Radius + Robot_MouthRadius +
+                                  *_bufferDistBeforeContact);
+    Point noisyTargetVel =
+        approachDir * (ball.vel.dot(approachDir) + *_touchDeltaSpeed);
```

Comment:
I was using a ball velocity average on the original code. I think if the touchDeltaSpeed was low enough, you could see the noise being the main determination of approach velocity. This will be something you have to test on a real field.